abi.go

```diff
@@ -15,20 +15,26 @@ import (
 
 // MapABI are the attributes of a Map which are available across all supported kernels.
 type MapABI struct {
-	Type       MapType
-	KeySize    uint32
-	ValueSize  uint32
-	MaxEntries uint32
-	Flags      uint32
+	Type          MapType
+	ID            MapID
```

Comment:
IMO name, ID aren't part of the ABI. They don't affect the behaviour of the map after all. Not sure about OwnerProgType, what is it used for?

Reply:
So where would you propose to expose them? Given the godoc comment of `MapABI` ("MapABI are the attributes of a Map which are available across all supported kernels.") I'd actually consider all of the fields part of the ABI, no? What would be your definition of the ABI?

Reply:
Yeah, the documentation is terrible, sorry. I've been struggling to describe it better.

In my mind, the ABI tells you whether a map is compatible with your code. Say, if you expect an LPM map, you can verify that you have been passed one. Same for the key and value sizes, etc. For this reason, I want to keep MapABI the same across all the supported kernels (currently >= 4.9).

ID and Name don't change the behaviour, from the point of view what operations you can perform on the map. Hence I wouldn't consider them to be part of the ABI. At the same time this seems to confuse a lot of people interacting with the library for the first time, so who knows.

Reply:
I see, thanks for explaining. That makes sense to me but it would probably be good to document this somewhere?